src/NuGet.Core/NuGet.Protocol/Converters/PackageDependencyGroupConverter.cs

```diff
@@ -17,11 +15,31 @@ public class PackageDependencyGroupConverter : JsonConverter
 
         public override bool CanWrite => false;
 
+        private readonly PackageDependencyConverter _converter = new PackageDependencyConverter();
+
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            var set = JsonUtility.LoadJson(reader);
-
-            var fxName = set.Value<string>(JsonProperties.TargetFramework);
+            string fxName = null;
+            List<Packaging.Core.PackageDependency> packages = new List<Packaging.Core.PackageDependency>();
+            while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+            {
+                if (reader.TokenType.Equals(JsonToken.PropertyName))
+                {
+                    if (reader.Value.Equals(JsonProperties.Dependencies))
+                    {
+                        // Dependencies are stored in an array
+                        while (reader.Read() && reader.TokenType != JsonToken.EndArray)
+                        {
+                            Packaging.Core.PackageDependency package = (Packaging.Core.PackageDependency)_converter.ReadJson(reader, typeof(Packaging.Core.PackageDependency), null, serializer);
+                            packages.Add(package);
+                        }
+                    }
+                    else if (reader.Value.Equals(JsonProperties.TargetFramework))
+                    {
+                        fxName = reader.ReadAsString();
+                    }
```

Comment:
The same issue exists here as with the other converter. I think the best practice should be to [skip](https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_JsonReader_Skip.htm) any properties that we don't expect.